tests/test_bookmarks.py

```diff
@@ -0,0 +1,194 @@
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks').get(stream)
+
+                if replication_method == self.INCREMENTAL:
+                    replication_key = self.expected_replication_keys().get(stream).pop()
+                    bookmark_key = self.get_bookmark_key(stream)  # BUG_SRCE-4609
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_key_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_key_value)
+
+                    # bookmarked states (actual values)
+                    first_bookmark_value = first_bookmark_key_value.get(bookmark_key)
+                    second_bookmark_value = second_bookmark_key_value.get(bookmark_key)
+                    # bookmarked values as utc for comparing against records
+                    first_bookmark_value_utc = self.convert_state_to_utc(first_bookmark_value)
+                    second_bookmark_value_utc = self.convert_state_to_utc(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    self.assertEqual(second_bookmark_value, first_bookmark_value) # assumes no changes to data during test
+
+                    # Verify the second sync records respect the previous (simulated) bookmark value
+                    simulated_bookmark_value = new_state['bookmarks'][stream][bookmark_key]
+                    for message in second_sync_messages:
+                        replication_key_value = message.get('data').get(replication_key)
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
+                                                msg="Second sync records do not repect the previous bookmark.")
+
+                    # Verify the first sync bookmark value is the max replication key value for a given stream
+                    for message in first_sync_messages:
+                        replication_key_value = message.get('data').get(replication_key)
+                        self.assertLessEqual(replication_key_value, first_bookmark_value_utc,
+                                             msg="First sync bookmark was set incorrectly, a record with a greater rep key value was synced")
+
+                    # Verify the second sync bookmark value is the max replication key value for a given stream
+                    for message in second_sync_messages:
+                        replication_key_value = message.get('data').get(replication_key)
+                        self.assertLessEqual(replication_key_value, second_bookmark_value_utc,
+                                             msg="Second sync bookmark was set incorrectly, a record with a greater rep key value was synced")
```

Comment:
Could this be combined with the previous loop over `second_sync_messages`?

Reply:
Making change.